Confusing diagnostic on `?` operator: needs to implement `Into<Error>`

### Code

```Rust
#[derive(Debug)]
struct Error {}

struct AnotherError {}

impl Into<Error> for AnotherError {
    fn into(self) -> Error {
        Error {}
    }
}

fn main() -> Result<(), Error> {
    let r: Result<(), AnotherError> = Err(AnotherError {});
    r?;
    Ok(())
}
```

### Current output

```Shell
Compiling example v0.1.0 (...)
error[E0277]: `?` couldn't convert the error to `Error`
  --> src/main.rs:20:6
   |
13 | fn main() -> Result<(), Error> {
   |              ----------------- expected `Error` because of this
14 |     let r: Result<(), AnotherError> = Err(AnotherError {});
15 |     r?;
   |     -^ the trait `From<AnotherError>` is not implemented for `Error`
   |     |
   |     this can't be annotated with `?` because it has type `Result<_, AnotherError>`
   |
note: `Error` needs to implement `From<AnotherError>`
  --> src/main.rs:2:1
   |
 2 | struct Error {}
   | ^^^^^^^^^^^^
note: alternatively, `AnotherError` needs to implement `Into<Error>`
  --> src/main.rs:4:1
   |
 4 | struct AnotherError {}
   | ^^^^^^^^^^^^^^^^^^^
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
```

### Desired output

```Shell
Compiling example v0.1.0 (...)
error[E0277]: `?` couldn't convert the error to `Error`
  --> src/main.rs:20:6
   |
13 | fn main() -> Result<(), Error> {
   |              ----------------- expected `Error` because of this
14 |     let r: Result<(), AnotherError> = Err(AnotherError {});
15 |     r?;
   |     -^ the trait `From<AnotherError>` is not implemented for `Error`
   |     |
   |     this can't be annotated with `?` because it has type `Result<_, AnotherError>`
   |
note: `Error` needs to implement `From<AnotherError>`
  --> src/main.rs:2:1
   |
 2 | struct Error {}
   | ^^^^^^^^^^^^
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
```

### Rationale and extra context

The compiler should not mention that "alternatively, `AnotherError` needs to implement `Into<Error>`", as it is not sufficient to make the `?` operator work. In this particular case, we already implemented `Into<Error>` for `AnotherError`, which makes the message even more confusing (we get the same output without that implementation).

### Other cases

```Rust

```

### Rust Version

```Shell
$ rustc --version --verbose
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-unknown-linux-gnu
release: 1.92.0
LLVM version: 21.1.3
```

### Anything else?

_No response_

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_trait_selection/src/error_reporting/traits/fulfillment_errors.rs
+++ b/compiler/rustc_trait_selection/src/error_reporting/traits/fulfillment_errors.rs
@@ -1287,10 +1287,6 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
                     self.tcx.def_span(def.did()),
                     format!("`{self_ty}` needs to implement `From<{ty}>`"),
                 );
-                err.span_note(
-                    self.tcx.def_span(found.did()),
-                    format!("alternatively, `{ty}` needs to implement `Into<{self_ty}>`"),
-                );
             }
             (ty::Adt(def, _), None) if def.did().is_local() => {
                 let trait_path = self.tcx.short_string(

--- a/tests/ui/try-trait/bad-question-mark-on-trait-object.rs
+++ b/tests/ui/try-trait/bad-question-mark-on-trait-object.rs
@@ -1,6 +1,5 @@
 struct E;
 //~^ NOTE `E` needs to implement `std::error::Error`
-//~| NOTE alternatively, `E` needs to implement `Into<X>`
 struct X; //~ NOTE `X` needs to implement `From<E>`
 
 fn foo() -> Result<(), Box<dyn std::error::Error>> { //~ NOTE required `E: std::error::Error` because of this

--- a/tests/ui/try-trait/bad-question-mark-on-trait-object.stderr
+++ b/tests/ui/try-trait/bad-question-mark-on-trait-object.stderr
@@ -1,5 +1,5 @@
 error[E0277]: `?` couldn't convert the error: `E: std::error::Error` is not satisfied
-  --> $DIR/bad-question-mark-on-trait-object.rs:7:13
+  --> $DIR/bad-question-mark-on-trait-object.rs:6:13
    |
 LL | fn foo() -> Result<(), Box<dyn std::error::Error>> {
    |             -------------------------------------- required `E: std::error::Error` because of this
@@ -17,7 +17,7 @@ LL | struct E;
    = note: required for `Box<dyn std::error::Error>` to implement `From<E>`
 
 error[E0277]: `?` couldn't convert the error to `X`
-  --> $DIR/bad-question-mark-on-trait-object.rs:18:13
+  --> $DIR/bad-question-mark-on-trait-object.rs:17:13
    |
 LL | fn bat() -> Result<(), X> {
    |             ------------- expected `X` because of this
@@ -27,14 +27,9 @@ LL |     Ok(bar()?)
    |        this can't be annotated with `?` because it has type `Result<_, E>`
    |
 note: `X` needs to implement `From<E>`
-  --> $DIR/bad-question-mark-on-trait-object.rs:4:1
+  --> $DIR/bad-question-mark-on-trait-object.rs:3:1
    |
 LL | struct X;
-   | ^^^^^^^^
-note: alternatively, `E` needs to implement `Into<X>`
-  --> $DIR/bad-question-mark-on-trait-object.rs:1:1
-   |
-LL | struct E;
    | ^^^^^^^^
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
 

